demosplan/DemosPlanCoreBundle/ResourceTypes/OriginalStatementResourceType.php

```diff
@@ -67,18 +81,77 @@ protected function getAccessConditions(): array
 
     public function isGetAllowed(): bool
     {
-        return $this->currentUser->hasPermission('feature_json_api_original_statement');
+        return $this->hasAccessPermissions();
     }
 
     public function isListAllowed(): bool
     {
-        return $this->currentUser->hasPermission('feature_json_api_original_statement');
+        return $this->hasAccessPermissions();
     }
 
-    protected function getProperties(): array
+    protected function getProperties(): ResourceConfigBuilderInterface
     {
-        return [
-            $this->createIdentifier()->readable()->filterable(),
-        ];
+        $originalStatementConfig = $this->getConfig(OriginalStatementResourceConfigBuilder::class);
+        $originalStatementConfig->id->setReadableByPath()->SetFilterable();
+        $originalStatementConfig->externId->setReadableByPath();
+        $originalStatementConfig->meta->setRelationshipType(
+            $this->getTypes()->getStatementMetaResourceType()
+        )->setReadableByPath();
+        $originalStatementConfig->submitDate->setReadableByPath()->setAliasedPath(Paths::statement()->submit);
+        $originalStatementConfig->isSubmittedByCitizen
+            ->setReadableByCallable(static fn (Statement $statement): bool => $statement->isSubmittedByCitizen());
+        $originalStatementConfig->fullText->setReadableByPath()->setAliasedPath(Paths::statement()->text);
+        $originalStatementConfig->shortText->setReadableByCallable(
+            static fn (Statement $statement): string => $statement->getTextShort()
+        )->setAliasedPath(Paths::statement()->text);
+        $originalStatementConfig->textIsTruncated
+            ->setReadableByCallable(static fn (Statement $statement): bool => $statement->getText() !== $statement->getTextShort());
+        $originalStatementConfig->phase
+        ->setReadableByCallable(
+            fn (Statement $statement): string => $this->statementService->getInternalOrExternalPhaseNameFromObject($statement)
```

Comment:
I see in the AbstractStatementResourceType this code (below) I was thinking we can use that same code or we move it to one place?
`\DemosEurope\DemosplanAddon\ResourceConfigBuilder\BaseStatementResourceConfigBuilder::@$phase`
```
        $configBuilder->phase
            ->readable(true, function (Statement $statement): string {
                $phase = $statement->getPhase();
                if (Statement::INTERNAL === $statement->getPublicStatement()) {
                    $internalPhases = $this->globalConfig->getInternalPhasesAssoc();
                    $phase = $internalPhases[$phase]['name'] ?? '';
                } else {
                    $externalPhases = $this->globalConfig->getExternalPhasesAssoc();
                    $phase = $externalPhases[$phase]['name'] ?? '';
                }

                return $phase;
            });
```

Reply:
What do you think if we use this method in both places?
[3d68a94](https://github.com/demos-europe/demosplan-core/pull/3826/commits/3d68a9487c5b5c0815d7dcba7641d3a0a623d357)

Reply:
Good idea, if it gives the same result then it should be good

Reply:
Its not the same result. We decided to use the approach from the ResourceType. See [0c4714b](https://github.com/demos-europe/demosplan-core/pull/3826/commits/0c4714b4ed0a6ffd87f74369fe0745e826bbc4b2) and [3aa149c](https://github.com/demos-europe/demosplan-core/pull/3826/commits/3aa149c3d0304ee1b727347b8e9bff55031e645a)